lib/appsignal/hooks/dry_monitor.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Appsignal
+  class Hooks
+    # @api private
+    class DryMonitorHook < Appsignal::Hooks::Hook
+      register :dry_monitor
+
+      def dependencies_present?
+        defined?(::Dry::Monitor::Notifications)
+      end
+
+      def install
+        require "appsignal/integrations/dry_monitor"
+
+        ::Dry::Monitor::Notifications.send(:prepend, Appsignal::Integrations::DryMonitorIntegration)
```

Comment:
I know we went this route for the active support notifications integration because of performance issues, but I don't know if that's a problem here. We can instead subscribe to the wildcard `*` event name instead to receive events from the instrumentation and use the built-in method of listening for events.

Reply:
If I recall correctly (I was taking a look at this yesterday with Luismi) users can create _several_ `Dry::Monitor::Notifications` instances, each being its own "notification hub" of sorts -- it's not a singleton like `ActiveSupport::Notifications`. This allows us to instrument all instances.